src/components/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,145 @@
+                            disabled || readOnly ? 'tw-text-black-60' : 'tw-text-black-80',
+                        ])}
+                        data-test-id={`${dataTestId}-decorator`}
+                    >
+                        {decorator}
+                    </div>
+                ) : null}
+                <input
+                    {...props}
+                    id={useMemoizedId(propId)}
+                    ref={inputElement}
+                    name="fondue-number-input"
+                    type={numberInputType}
+                    className={merge([
+                        'tw-w-full tw-grow tw-border-none tw-outline-none tw-bg-transparent tw-hide-input-arrows',
+                        disabled || readOnly
+                            ? 'tw-text-black-40 tw-placeholder-black-30 dark:tw-text-black-30 dark:tw-placeholder-black-40'
+                            : 'tw-text-black tw-placeholder-black-60 dark:tw-text-white',
+                    ])}
+                    onClick={() => inputElement.current?.focus()}
+                    onChange={(event) => handleOnChange(event.currentTarget.value)}
+                    data-test-id={`${dataTestId}-input`}
+                    onBlur={onBlur}
+                    onKeyDown={onKeyDown}
+                    required={required}
+                    readOnly={readOnly}
+                    disabled={disabled}
+                    onFocus={(event) => (onFocus ? onFocus(event) : null)}
+                    size={size}
+                />
+                {!!incrementable ? (
+                    <>
+                        <button
+                            type={buttonInputType}
+                            onClick={() => {
+                                !!inputElement.current &&
+                                    handleCount(inputElement.current.value, NumberInputIncrement.DECREMENT);
+                            }}
+                            data-test-id={`${dataTestId}-decrement`}
+                        >
+                            <IconMinus16 />
+                        </button>
+                        <button
+                            type={buttonInputType}
+                            onClick={() => {
+                                !!inputElement.current &&
+                                    handleCount(inputElement.current.value, NumberInputIncrement.INCREMENT);
+                            }}
+                            data-test-id={`${dataTestId}-increment`}
+                        >
+                            <IconPlus16 />
+                        </button>
+                    </>
+                ) : null}
+
+                {error ? (
+                    <span className="tw-text-red-60">
+                        <IconExclamationMarkTriangle16 data-test-id={`${dataTestId}-error-icon`} />
+                    </span>
+                ) : null}
+            </div>
+            {error && errorText ? (
```

Comment:
Should errorText or let say HelperText be part of the component itself or the new `FormField` component?

Reply:
Good question.  Initially I thought `FormField` but was pulled back to in the input as all inputs should require some validation, so feels like a low level item.  But happy to remove it from here and add it back when to new `FormField` is built.